CypherpunkArmory/UserLAnd
# Download assets from github releases

**Describe the pull request**
This PR switches our download implementation to use the releases section of Github. This is to prevent asset corruption that was found when downloading assets directly from the git tree. 

The new flow queries the Github releases API to find endpoints for the assets list text file and the assets themselves, which are now released as tar.gz archives. Therefore:

1. Query API to find assets list text file
2. Read assets list directly into memory
3. Query latest release version
4. Download newest archives as necessary
5. Unpack newest archives as necessary
6. Verify unpacked assets match assets list

A good to place to start reviewing could be the changes to the session FSM. All steps from retrieving the assets list to verifying assets have been touched.

**Link to relevant issues**
This PR is intended to fix many of the issues we've seen cropping up around download issues, filesystem, and asset corruption.

Potentially affects #703, #692, #685, #634, #566, #486, #462, #461, #409, #380, #364.
